src/secret.rs

```diff
@@ -0,0 +1,28 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use std::str::FromStr;
+
+use anyhow::{Context, Error};
+
+#[derive(Debug)]
+pub struct SecretFile(String);
+
+impl From<SecretFile> for String {
+    fn from(sf: SecretFile) -> Self {
+        sf.0
+    }
+}
+
+impl FromStr for SecretFile {
+    type Err = Error;
+
+    fn from_str(path: &str) -> Result<Self, Self::Err> {
```

Comment:
I'm not sure this is correct usage of the trait, it would enable `SecretFile::parse("/some/path")`. AFAIK typically these would transform `/some/path` and not "dereference" it

Reply:
It is somewhat abusive, yes. OTOH, it is the only way to actually inject this logic into `clap`. I chose this trade-off because `SecretFile` is far more self-documenting than polluting an already busy initialization function.